### PR TITLE
HTML: test rendering of legend with different display values

### PR DESCRIPTION
**Pull request by zcorpan**
See https://github.com/whatwg/html/pull/3934

### CODE REVIEW

File: html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-rendering-ref.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+<div class=padding>table-column</div>
+<div>table-caption</div>
+<div>flow</div>
+<div>flow-root</div>
+<div>run-in</div>
+<div>inline</div>
+<div>inline-block</div>
+<div><span class=cell>inline-table</span><span class=cell>inline-table</span></div>
+<div><span class=ruby>ruby</span><span class=ruby>ruby</span></div>
+<div>ruby-base</div>
```
Comment (mstensho) [inline html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-rendering-ref.html]: What's these four lines supposed to test? Blink passes, although it has no support for ruby display types.

Comment (zcorpan) [reply html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-rendering-ref.html]: That they blockify. Blink fails the "ruby" test.

Comment (mstensho) [reply html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-rendering-ref.html]: Right. Guess there's no real point in making these fail for engines (like Blink) that don't support these display property values at all. So, fine.

Comment (zcorpan) [reply html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-rendering-ref.html]: Yeah. There's a separate test (legend-display.html) that fails when the value is not supported.

File: html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-rendering.html [hunk 2]
```diff
@@ -0,0 +1,119 @@
+<!doctype html>
+<title>rendered legend and CSS display rendering</title>
+<link rel=match href=legend-display-rendering-ref.html>
+<style>
+body { margin: 0; }
+fieldset { margin: 0; padding: 0; border: none; }
+legend { width: 600px; box-sizing: border-box; padding: 0; background: #ddd; border: 1px solid; }
+[style="display: table"] span,
+[style="display: inline-table"] { display: table-cell; }
```
[the remark this answers is not in the document]
Comment (zcorpan) [reply html/rendering/non-replaced-elements/the-fieldset-and-legend-elements/legend-display-rendering.html]: oops, fixed.

**Review by mstensho — APPROVED** [review]

**Pull request merged**